builtin/credential/aws-ec2/path_login.go

```diff
@@ -244,24 +244,28 @@ func (b *backend) pathLoginUpdate(
 		return logical.ErrorResponse("role entry not found"), nil
 	}
 
-	// Only 'bound_ami_id' and 'bound_iam_role_arn' constraints are supported on the role currently.
-	// Check if the AMI ID of the instance trying to login matches the
+	// Verify that the AMI ID of the instance trying to login matches the
 	// AMI ID specified as a constraint on the role.
 	if roleEntry.BoundAmiID != "" && identityDoc.AmiID != roleEntry.BoundAmiID {
-		return logical.ErrorResponse(fmt.Sprintf("AMI ID %s does not belong to role %s", identityDoc.AmiID, roleName)), nil
+		return logical.ErrorResponse(fmt.Sprintf("AMI ID '%s' does not belong to role '%s'", identityDoc.AmiID, roleName)), nil
+	}
+
+	// Verify that the AccountID of the instance trying to login matches the
+	// AccountID specified as a constraint on the role.
+	if roleEntry.BoundAccountID != "" && identityDoc.AccountID != roleEntry.BoundAccountID {
+		return logical.ErrorResponse(fmt.Sprintf("Account ID '%s' does not belong to role '%s'", identityDoc.AccountID, roleName)), nil
 	}
 
 	// Check if the IAM Role ARN of the instance trying to login matches the
 	// IAM Role ARN specified as a constraint on the role.
 	if roleEntry.BoundIamARN != "" {
 		if instanceDesc.Reservations[0].Instances[0].IamInstanceProfile == nil {
-			return nil, fmt.Errorf("Iam instance profile in instance description is nil")
+			return nil, fmt.Errorf("IamInstanceProfile in the instance description is nil")
 		}
 		if instanceDesc.Reservations[0].Instances[0].IamInstanceProfile.Arn == nil {
-			return nil, fmt.Errorf("ARN in instance description is nil")
+			return nil, fmt.Errorf("Arn in the instance description is nil")
```

Comment:
Same here -- I know that you're trying to give the 'actual value' but most people aren't going to see that API output anyways. I'd just use ARN as otherwise someone seeing this error will think it's wonky.